server/controllers/users.js

```diff
@@ -90,12 +92,125 @@ class UserController {
   }
 
   /**
-   * Signuout user and blacklist tokens
+   * Users can follow each ither
    * @param {object} req
    * @param {object} res
    * @param {object} next
    * @returns {object} res message
    */
+  static async follow(req, res, next) {
+    try {
+      const { username } = req.params;
+      const { id } = req.decoded;
+
+      const userToFollow = await User.findOne({
+        where: {
+          username,
+        }
+      });
+      if (!userToFollow) {
+        return Response.error(res, 404, 'User not found');
+      }
+      const followid = userToFollow.dataValues.id;
+      if (followid === id) {
+        return Response.error(res, 400, 'you cannot follow yourself');
+      }
+      const [followed, created] = await Follower.findOrCreate({
```

Comment:
This could be expressed some other way.

please look through the sequelize documentation

`user.addFollower(req.user)`
(user here is the user to be followed)

You may want to modify the `Follower` model.

change `followingId` to `userId` 

having `follower` and `following` makes very little sense 
either someone is being followed or someone is following someone

in our case, would be nice to have a user being followed (so a user has followers)

Reply:
done eventually.